### ISSUE TITLE
If request starts after the signal has already been aborted, it won't throw an error

### ISSUE THREAD

**Issue by purpurplr** [sindresorhus/ky#660]
If request starts after the signal has already been aborted, it won't throw an error, although it probably should.

```
const controller = new AbortController();

controller.abort('no reason');

ky.get('https://jsonplaceholder.typicode.com/todos/1', { signal: controller.signal })
  .json()
  .then((json) => console.log(json));
```

**Comment by sindresorhus** [comment]
Agreed

**Comment by sholladay** [comment]
Agreed.

Does it actually send the request or does it silently fail? Also, what does fetch do?

**Comment by purpurplr** [comment]
Ky sends the request, fetch throws an error.

**Comment by sholladay** [comment]
I believe this is happening because when we [create our about controller](https://github.com/sindresorhus/ky/blob/3ba40cc6333cf1847c02c51744e22ab7c04407f5/source/core/Ky.ts#L167), we only listen for future aborts from the user's signal.

We need to either copy `originalSignal`'s existing state or use `throwIfAborted()` on it.

**Comment by kadhirr** [comment]
Hey folks, I prepared a PR for this (with `throwIfAborted()`), can I get this issue assigned to me? I am facing an issue here with the tests though. Running tests seem to just throw the following error:
```
  TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".ts" for D:\code\github\ky\test\prefix-url.ts

  ✘ test\http-error.ts exited with a non-zero exit code: 1
  ✘ test\main.ts exited with a non-zero exit code: 1
  ✘ test\methods.ts exited with a non-zero exit code: 1
  ✘ test\prefix-url.ts exited with a non-zero exit code: 1

  Uncaught exception in test\retry.ts

  TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".ts" for D:\code\github\ky\test\retry.ts

  ✘ test\retry.ts exited with a non-zero exit code: 1
  ```
  Has anyone faced this issue?

**Comment by sholladay** [comment]
I'm not sure if the tests have been run on Windows before.

My recommendation would be to use the Windows Subsystem for Linux, which is the official way to run *nix software on Windows. But we would probably also accept a patch to get the tests working on Windows, assuming it's something simple like a file path issue.

**Comment by kadhirr** [comment]
Added a PR for this [here](https://github.com/sindresorhus/ky/pull/663). This is my first time helping out a open source library, please let me know how to improve my PR and I will be happy to contribute.

I tried to add a test for this scenario. Facing some issues there, would the right place to discuss this be this issue or the PR itself?

**Comment by sholladay** [comment]
Fixed in #663
